spark/src/main/java/org/apache/iceberg/actions/Actions.java

```diff
@@ -79,7 +80,8 @@ public RewriteDataFilesAction rewriteDataFiles() {
   }
 
   public ExpireSnapshotsAction expireSnapshots() {
-    return new ExpireSnapshotsAction(spark, table);
+    BaseExpireSnapshotsSparkAction delegate = new BaseExpireSnapshotsSparkAction(spark, table);
+    return new ExpireSnapshotsAction(delegate);
```

Comment:
Why do we use a deprecated class(ExpireSnapshotsAction) here?

Reply:
This is a user-facing API so we cannot break it without deprecating first.

Reply:
Not sure about the Iceberg deprecating process. Here is my understanding, we need to create a new method with new return type for this user-facing API, which can be used by user moving forward. Meanwhile we deprecate this method.

Reply:
Oh, yeah, we will mark `Actions` deprecated once we have a new alternative. We will have a totally new entry point called `SparkActions` in another package. Right now, I am migrating action by action to reduce the amount of changes in a single pr. Once all actions are done, I'll create a new public API for users to use and deprecate the rest.